specifyweb/frontend/js_src/lib/components/AttachmentsBulkImport/SelectUploadPath.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+import { attachmentsText } from '../../localization/attachments';
+import { syncFieldFormat } from '../../utils/fieldFormat';
+import { defined } from '../../utils/types';
+import { Select } from '../Atoms/Form';
+import { strictGetModel } from '../DataModel/schema';
+import { QueryFieldSpec } from '../QueryBuilder/fieldSpec';
+import type { PartialAttachmentUploadSpec } from './Import';
+import { staticAttachmentImportPaths } from './importPaths';
+
+export function SelectUploadPath({
+  onCommit: handleCommit,
+  currentKey,
+}: {
+  readonly onCommit:
+    | ((commitableSpec: PartialAttachmentUploadSpec) => void)
+    | undefined;
+  readonly currentKey: keyof typeof staticAttachmentImportPaths | undefined;
+}): JSX.Element {
+  const [staticKey, setStaticKey] = React.useState<
+    keyof typeof staticAttachmentImportPaths | undefined
+  >(currentKey);
+  function handleBlur(): void {
+    if (staticKey === currentKey || staticKey === undefined || staticKey === '')
+      return;
+    handleCommit?.(generateUploadSpec(staticKey));
+  }
+
+  return (
```

Comment:
I really don't like the "Choose Path" UX.
Could you put it right above the table - just like in workbench import, not part of the header (or alternatively, make it aligned right so that it's near the save button)? It's required to initialize the upload, but being in the header makes it seem like an optional thing

Reply:
oh, I didn't realize that it disappears as soon as you select the value. then there is absolutely no reason for this Select box not to be in much more central place - right above the table (or where the table would be if there is no table yet)

Reply:
![image](https://github.com/specify/specify7/assets/40512816/b6e6b391-123b-420e-8f3b-b7a0b9f9679f)

nevermind. it doesn't disapper. it just get's "consumed" by other elements from lack of space. Your forgot to add "flex-wrap" class name. Would look much better:

![Screenshot 2023-10-02 at 19 57 37](https://github.com/specify/specify7/assets/40512816/680568e6-b3a4-4f52-a90d-d3faafb09622)

Reply:
:joy: